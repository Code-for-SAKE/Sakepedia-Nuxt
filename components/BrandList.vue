<template>
  <div>
    <b-pagination
      v-if="brands.length > 2"
      v-model="page"
      :total-rows="count * limit"
      :per-page="limit"
      prev-text="Prev"
      next-text="Next"
      align="center"
      @change="handlePageChange"
    />
    <div v-if="brands.length" class="list-group">
      <nuxt-link
        v-for="brand in brands"
        :key="brand._id"
        class="list-group-item list-group-item-action"
        :to="'/brands/' + brand._id"
      >
        {{ brand.name }}
      </nuxt-link>
    </div>

    <div v-else class="alert alert-warning">データがありません</div>
  </div>
</template>

<script>
import { getList } from "../lib/ApiClient/getList";
export default {
  props: {
    search: {
      type: Object,
      default: () => {},
    },
  },
  data() {
    return {
      brands: [],

      search_: this.search,

      page: 1,
      count: 0,
      limit: 10,
    };
  },
  created() {
    getList("brands", { brewery: this.search.brewery }).then((res) => {
      this.brands = res.list;
      this.page = res.currentPage;
      this.count = res.count;
    });
  },
  methods: {
    async retrieves() {
      const { list, currentPage, count } = await getList("brands", {
        page: this.page,
        limit: this.limit,
      });
      this.brands = list;
      this.page = currentPage;
      this.count = count;
    },
    handlePageChange(value) {
      this.page = value;
      this.retrieves();
    },
  },
};
</script>
