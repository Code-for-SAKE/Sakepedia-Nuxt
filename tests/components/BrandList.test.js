import { mount, RouterLinkStub } from '@vue/test-utils'
import BrandList from '@/components/BrandList.vue'

let wrapper;
const responseMock = {
  data: {
    brands: [
      {_id: 'test', name: 'test'},
      {_id: 'test2', name: 'test2'},
      {_id: 'test3', name: 'test3'}
    ],
    currentPage: 1,
    pageCount: 0
  }
}
beforeAll(() => {
  wrapper = mount(BrandList, {
    mocks: {
      $axios: {
        get: jest.fn(() => Promise.resolve(responseMock))
      }
    },
    props: {
      search: {}
    },
    stubs: {
      'nuxt-link': RouterLinkStub,
      'b-pagination': true
    }
  });
});

describe('components/BrandList.vue', () => {
  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });
  describe('template', () => {
    describe('will be rendered correctly', () => {
      it('with data', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });
      it('without data', () => {
        wrapper = mount(BrandList, {
          mocks: {
            $axios: {
              get: jest.fn(() => Promise.resolve({
                data: {
                  brands: [],
                  currentPage: 1,
                  pageCount: 0
                }
              }))
            }
          },
          props: {
            search: {}
          }
        });
        expect(wrapper.html()).toMatchSnapshot();
      });
    });
  });
  describe('script', () => {
    describe('methods', () => {
      describe('getRequestParams', () => {
        it('return correct value', () => {
          const page = 1;
          const limit = 10;
          wrapper.vm.search_ = {}
          expect(wrapper.vm.getRequestParams(page,limit)).toEqual({
            params: {
              page: page,
              limit: limit
            }
          });
        });
      });
      describe('retrieves', () => {
        const page = 1
        const limit = 10
        it('getRequestParams has been called', () => {
          const spyGetRequestParams = jest.spyOn(wrapper.vm, 'getRequestParams');
          wrapper.vm.retrieves();
          expect(spyGetRequestParams).toHaveBeenCalledWith(page,limit);
        });
        it('$axios.get has been called', () => {
          const spyAxiosGet = jest.spyOn(wrapper.vm.$axios, 'get')
          expect(spyAxiosGet).toHaveBeenCalledWith('/api/brands', { params: { page: page, limit: limit } });
        });
      });
      describe('handlePageChange', () => {
        it('retrieves() has been called', () => {
          const spyRetrieves = jest.spyOn(wrapper.vm, 'retrieves');
          wrapper.vm.handlePageChange();
          expect(spyRetrieves).toHaveBeenCalled();
        });
      });
    });
  });
});
