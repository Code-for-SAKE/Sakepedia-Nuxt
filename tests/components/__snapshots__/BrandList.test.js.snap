// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BrandList.vue template renderd correctly with data will be renderd correctly 1`] = `
"<div>
  <b-pagination-stub total-rows=\\"0\\" per-page=\\"10\\" prev-text=\\"Prev\\" next-text=\\"Next\\" align=\\"center\\" value=\\"1\\"></b-pagination-stub>
  <div class=\\"list-group\\"><a class=\\"list-group-item list-group-item-action\\">
      test
    </a><a class=\\"list-group-item list-group-item-action\\">
      test2
    </a><a class=\\"list-group-item list-group-item-action\\">
      test3
    </a></div>
</div>"
`;

exports[`components/BrandList.vue template renderd correctly without data will be renderd correctly 1`] = `
"<div>
  <!---->
  <div class=\\"alert alert-warning\\">データがありません</div>
</div>"
`;
